lib/ansible/plugins/action/reboot.py

```diff
@@ -412,7 +412,7 @@ def run(self, tmp=None, task_vars=None):
             return {'changed': False, 'elapsed': 0, 'rebooted': False, 'failed': True, 'msg': msg}
 
         if self._play_context.check_mode:
-            return {'changed': True, 'elapsed': 0, 'rebooted': True}
+            raise AnsibleActionSkip('Check mode is not supported for this task.')
```

Comment:
We know reboot will always make a change so there's no need to disable check mode support. It's accurately reporting a change would occur. You can even see the intention a few lines above: `self._supports_check_mode = True`.